### PR TITLE
[MIST-1030] Separate fields from ApplicationInfo level to Group level

### LINKED ISSUE #1030: Separate fields from ApplicationInfo level to Group level
Currently, the DefaultApplicationInfoImpl class has fields that are not app-specific.
These should be moved to the Group level.

These are:
- `QueryStarter`
- `QueryRemover`
- `ExecutionDags`

### PR DESCRIPTION
**Pull request by kyungissac**
This PR resolves #1030.

Only the minimum amount of code was changed to directly address #1030.
The complete separation of groups and applications in checkpointing is the next step.

Closes #1030

### CODE REVIEW

File: mist-core/src/main/java/edu/snu/mist/core/task/groupaware/Group.java [hunk 1]
```diff
@@ -149,4 +152,10 @@
    * The number of active queries.
    */
   int size();
+
+  QueryStarter getQueryStarter();
+
+  QueryRemover getQueryRemover();
+
```
Comment (DifferentSC) [inline mist-core/src/main/java/edu/snu/mist/core/task/groupaware/Group.java]: Please add comments on these three methods.

File: mist-core/src/main/java/edu/snu/mist/core/task/groupaware/GroupAwareQueryManagerImpl.java [hunk 2]
```diff
@@ -199,7 +206,9 @@ public Query createAndStartQuery(final String queryId,
     groupAllocationTableModifier.addEvent(new WritingEvent(WritingEvent.EventType.QUERY_ADD,
         new Tuple<>(applicationInfo, query)));
     // Start the submitted dag
-    applicationInfo.getQueryStarter().start(queryId, query, configDag, applicationInfo.getJarFilePath());
+    // TODO : [MIST-1016] get the appropriate Group for this applicationInfo.
+    final Group group = applicationInfo.getGroups().get(0);
```
Comment (DifferentSC) [inline mist-core/src/main/java/edu/snu/mist/core/task/groupaware/GroupAwareQueryManagerImpl.java]: How about using random select here?

**Comment by kyungissac** [comment]
@DifferentSC I've addressed your comments. Thanks!

File: mist-core/src/main/java/edu/snu/mist/core/task/groupaware/GroupAwareQueryManagerImpl.java [hunk 3]
```diff
@@ -227,6 +238,8 @@ public ApplicationInfo createApplication(final String appId, final List<String>
     groupAllocationTableModifier.addEvent(
             new WritingEvent(WritingEvent.EventType.GROUP_ADD, new Tuple<>(applicationInfo, group)));
 
+    groupMap.putIfAbsent(groupId, group);
```
Comment (DifferentSC) [inline mist-core/src/main/java/edu/snu/mist/core/task/groupaware/GroupAwareQueryManagerImpl.java]: I think this should be processed in `SingleWriterThread`.

**Comment by DifferentSC** [comment]
Approved. I'm merging this.

**Pull request merged**